backend/app/models/sample_type.rb

```diff
@@ -0,0 +1,284 @@
+# sample_types table
+class SampleType < ActiveRecord::Base
+
+  validates :name,        presence: true, uniqueness: { case_sensitive: false }
+  validates :description, presence: true
+
+  # Return all sample types.
+  #
+  # @return all sample types
+  def self.find_all
+    sql = "
+      select * from sample_types order by name
+    "
+    SampleType.find_by_sql sql
+  end
+
+  def self.find_id(id)
```

Comment:
Please use ActiveRecord unless raw SQL has some performance advantage.
This is equivalent to `SampleType.find_by(id: id)`